cmd/tk/util.go

```diff
@@ -15,31 +15,36 @@ func pageln(i ...interface{}) {
 	fPageln(strings.NewReader(fmt.Sprint(i...)))
 }
 
-// fPageln invokes the systems pager with the supplied data
-// falls back to fmt.Println() when paging fails or non-interactive
+// fPageln invokes the systems pager with the supplied data.
+// If the PAGER environment variable is empty, no pager is used.
+// If the PAGER environment variable is unset, use less with posix flags.
 func fPageln(r io.Reader) {
-	// get system pager, fallback to `less`
-	pager := os.Getenv("PAGER")
-	var args []string
-	if pager == "" || pager == "less" {
+	pagerEnv, ok := os.LookupEnv("PAGER")
+	if !ok {
 		// --RAW-CONTROL-CHARS  Honors colors from diff. Must be in all caps, otherwise display issues occur.
 		// --quit-if-one-screen Closer to the git experience.
 		// --no-init            Don't clear the screen when exiting.
-		pager = "less"
-		args = []string{"--RAW-CONTROL-CHARS", "--quit-if-one-screen", "--no-init"}
+		pagerEnv = "less --RAW-CONTROL-CHARS --quit-if-one-screen--no-init"
 	}
 
-	// invoke pager
-	cmd := exec.Command(pager, args...)
-	cmd.Stdin = r
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	pager := strings.Split(pagerEnv, " ")
 
-	// if this fails, just print it
-	if err := cmd.Run(); err != nil {
-		if _, err = io.Copy(os.Stdout, r); err != nil {
-			log.Fatalln("Writing to Stdout:", err)
+	if interactive && len(pager) > 0 {
+		pagerCmd := pager[0]
+		pagerArgs := strings.Split(os.Getenv("PAGER"), " ")[1:len(pager)]
+
+		cmd := exec.Command(pagerCmd, pagerArgs...)
+		cmd.Stdin = r
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+
+		if err := cmd.Run(); err != nil {
+			if _, err = io.Copy(os.Stdout, r); err != nil {
+				log.Fatalln("Writing to Stdout:", err)
+			}
 		}
+	} else {
+		fmt.Println(r)
```

Comment:
Can't print an `io.Reader`, need to read first :)

Maybe put this the bottom of the func:

```go
_, err := io.Copy(os.Stdout, r)
return err
```

Then also when the cmd fails, don't return but instead just do "nothing", so it automatically falls back to above code

Reply:
Whoops! The proposed refactor sounds nice but changes the function signature for `fPageln` and potentially for `pageln` too. If that's cool, I'll go ahead and make this change.

Reply:
I'm happy with both `fPageln` and `pageln` returning errors, it should be up to the consumer to ignore them if applicable.

Also makes the control flow inside a bit simpler I guess